### PR TITLE
Pie 1376/attendance input errors

### PR DESCRIPTION
**Pull request by cjhaddad**
## 💅🏼 What issue does this fix?
this addresses #1376 and renders an error when a date for the column in not selected and needs to be.
WIP bc disabling of Save logic needs to be tweaked, but putting this PR up now for preliminary review

## 🍂 Before You Submit

* [ ] Did you write tests?
* [ ] Did you run Google Lighthouse and/or WebAIM (Wave) on UI components in your PR?
* [ ] Does your PR contain any required translations?
* [ ] Did you run `bundle exec rspec` from the root?
* [ ] Did you run `bundle exec rubocop` from the root?
* [x] Did you run `yarn lint` in `/client`?
* [x] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment

* [ ] Data Migrations
* [ ] Schema Migrations
* [ ] Dependencies

### CODE REVIEW

File: client/src/Attendance/Attendance.js [hunk 1]
```diff
@@ -84,15 +92,32 @@ export function Attendance() {
...
-            style={{ width: '8rem ', color: '#004A6E' }}
-          />
-        ),
+        title: () => {
+          const errorIsPresent =
+            Object.values(latestAttendanceData.current).find(
+              row => Object.keys(row[i]).length > 0
+            ) && latestColumnDates.current[i] === ''
+
+          if (errorIsPresent !== latestestDisableSave.current) {
```
Comment (katelovescode) [inline client/src/Attendance/Attendance.js:101]:
```suggestion
          if (errorIsPresent !== latestDisableSave.current) {
```

File: client/src/Attendance/Attendance.js [hunk 2]
```diff
@@ -84,15 +92,32 @@ export function Attendance() {
...
+            latestestDisableSave.current = errorIsPresent
```
Comment (katelovescode) [inline client/src/Attendance/Attendance.js:102]:
```suggestion
            latestDisableSave.current = errorIsPresent
```

**Review by katelovescode — COMMENTED** [review]
Minor typo stuff, I'll wait til it's done to do a functionality check

**Comment by katelovescode** [comment]
Couple of issues I found:

#1 - right after the page is loaded the SAVE button is active and is able to send requests to the API, which creates a parameter error in the backend.  The good news is, it doesn't stop the user from continuing, but we should probably prevent them from submitting with no data

save_no_data

#2 - if I get the "must select a date" error, and then pick the date, the Save button doesn't become active again....

save_inactive

...unless I make a change to another field

reactivated_save

From what I can tell these are the only error states I should be testing, based on the AC in the issue, let me know if I missed anything!

**Comment by katelovescode** [comment]
(also your PR still says WIP but I assume this is the one you wanted me to review!)

**Comment by cjhaddad** [comment]
ready for your eyeballs again @katelovescode

**Comment by katelovescode** [comment]
Looks great to me!

**Review by katelovescode — APPROVED** [review]

**Pull request merged**